packages/fuels-code-gen/src/program_bindings/resolved_type.rs

```diff
@@ -60,229 +53,217 @@ impl ToTokens for ResolvedType {
         tokens.extend(tokenized_type)
     }
 }
+
 impl Display for ResolvedType {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         write!(f, "{}", self.to_token_stream())
     }
 }
 
-/// Given a type, will recursively proceed to resolve it until it results in a
-/// `ResolvedType` which can be then be converted into a `TokenStream`. As such
-/// it can be used whenever you need the Rust type of the given
-/// `FullTypeApplication`.
-pub(crate) fn resolve_type(
-    type_application: &FullTypeApplication,
-    shared_types: &HashSet<FullTypeDeclaration>,
-) -> Result<ResolvedType> {
-    let recursively_resolve = |type_applications: &Vec<FullTypeApplication>| {
+pub(crate) struct TypeResolver {
+    relative_to_mod: TypePath,
+}
+
+impl TypeResolver {
+    pub(crate) fn new() -> Self {
```

Comment:
where you not advertising to use `new` with all fields as arguments and `default()` if you need the default values :)